main.tf

```diff
@@ -61,6 +79,29 @@ resource "azurerm_storage_account" "example" {
   location                 = azurerm_resource_group.example.location
   account_tier             = "Standard"
   account_replication_type = "GRS"
+
+  queue_properties {
+    hour_metrics {
+      version               = "1.0"
+      include_apis          = true
+      retention_policy_days = 365
+      enabled               = true
+    }
+    minute_metrics {
+      enabled               = true
+      version               = "1.0"
+      include_apis          = true
+      retention_policy_days = 365
+    }
+    logging {
+      delete                = true
```

Comment:
**Tenable.cs remediation**
 
**[MEDIUM]: Ensure that 'Auditing' Retention is 'greater than 90 days' for Azure MSSQL Server**
 
**Vulnerability**
Auditing for MSSQL Server can help with monitoring for errors or anomalies, create a record for compliance requirements, as well as aid with performance tuning of the environment. Many compliance frameworks and regulatory requirements have a minimum retention period of 90 days as a standard. For more information on auditing in MSSQL Server, see the Azure documentation.
References:
[https://learn.microsoft.com/en-us/azure/azure-sql/database/auditing-overview?view=azuresql](https://learn.microsoft.com/en-us/azure/azure-sql/database/auditing-overview?view=azuresql)
 
**Remediation**
In Azure Console -
1. Open the Azure Portal and go to SQL servers.
2. Choose the SQL server you wish to edit.
3. Under Auditing, Enable Azure SQL Auditing
4. Under Advanced properties, Set Retention (Days) to greater than 90 days
5. Select Save

In Terraform - 
1. In the azurerm_mssql_server_extended_auditing_policy resource, set retention_in_days to to greater than 90 days.

References:
[https://learn.microsoft.com/en-us/azure/azure-sql/database/auditing-overview?view=azuresql](https://learn.microsoft.com/en-us/azure/azure-sql/database/auditing-overview?view=azuresql)
[https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_server_extended_auditing_policy#retention_in_days](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_server_extended_auditing_policy#retention_in_days)

source: tenablecs-